### PR TITLE
Feature/complete mvp

### PR DESCRIPTION
**Pull request by Whitney-Rene**
This PR completes the MVP for my project.  My project has full functionality and some css.
Testing is also implemented in this branch.
I also edit, eliminate and add comments
I also install the libraries for testing and create 9 testing files.

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import RequestBouquet from './components/RequestBouquet';
 import NavBar from './components/NavBar';
 import EntryForm from './components/EntryForm';
 import Login from './components/Login';
-import './App.css'
+// import './App.css' //not sure I need this line of code??
```
Comment (plee-fn) [inline client/src/App.jsx:12]: If you don't need the styling, you can remove it.

File: client/src/__tests__/CreateRoseBudThorn.test.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { render, screen, mockRouter } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import CreateRoseBudThorn from '../components/CreateRoseBudThorn';
+
+test('CreateRoseBudThorn component renders buttons', () => {
+
+    render( 
+    <Router>
+        <CreateRoseBudThorn />
```
Comment (plee-fn) [inline client/src/__tests__/CreateRoseBudThorn.test.jsx:10]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/EditEntry.test.jsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { render, screen } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import EditEntry from '../components/EditEntry';
+
+test('buttons render', () => {
+  render(
+    <Router>
+        <EditEntry />
```
Comment (plee-fn) [inline client/src/__tests__/EditEntry.test.jsx:9]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/EntryForm.test.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import EntryForm from '../components/EntryForm';
+
+test('HomePage renders child components', () => {
+  render(
+    <Router>
+        <EntryForm />
```
Comment (plee-fn) [inline client/src/__tests__/EntryForm.test.jsx:9]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/HomePage.test.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { render, screen } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import HomePage from '../components/HomePage';
+
+test('HomePage renders child components', () => {
+  render(
+    <Router>
+        <HomePage />
```
Comment (plee-fn) [inline client/src/__tests__/HomePage.test.jsx:9]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/ListLatestEntries.test.jsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        },
+      ],
+    };
+  };
+});
+
+test('list entries on page', async () => {
+    render(
+    <Router>
+        <ListLatestEntries />
```
Comment (plee-fn) [inline client/src/__tests__/ListLatestEntries.test.jsx:29]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/Login.test.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { render, screen } from "@testing-library/react";
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import Login from "../components/Login";
+
+test('NavBar should render links case-insensitively', () => {
+    render(
+    <Router>
+        <Login />
```
Comment (plee-fn) [inline client/src/__tests__/Login.test.jsx:9]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/NavBar.test.jsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { render, screen } from "@testing-library/react";
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import NavBar from "../components/NavBar";
+
+test('NavBar should render links case-insensitively', () => {
+    render(
+    <Router>
+        <NavBar />
```
Comment (plee-fn) [inline client/src/__tests__/NavBar.test.jsx:9]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/RequestBouquet.test.jsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import { BrowserRouter as Router } from 'react-router-dom';
+// import userEvent from '@testing-library/user-event';
+
+
+import RequestBouquet from "../components/RequestBouquet";
+
+test ("buttons render", () => {
+    render(
+    <Router>
+        <RequestBouquet />
```
Comment (plee-fn) [inline client/src/__tests__/RequestBouquet.test.jsx:11]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

File: client/src/__tests__/app.test.jsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+// import userEvent from '@testing-library/user-event';
+
+import App from '../App';
+
+test ('App renders correctly', () => {
+
+    //reders the app component
+    render(
+        <Router>
+            <App />
```
Comment (plee-fn) [inline client/src/__tests__/app.test.jsx:12]: This is more of a general comment. Do you need to pass down props to this component to get it to work? Think about it!

**Review by plee-fn — COMMENTED** [review]
I realize that all of your components are not taking a prop parameter. One of the key features of React is being able to pass down props and reuse those props in children components. I'm wondering if we can somehow utilize this feature. It could save some redundant API calls to the backend and cleanup some code. Maybe we don't even need to do this? But it's good to think about it.

Besides this, I think the code looks good overall!

File: client/src/App.jsx (hunk 1 above)
Comment (Whitney-Rene) [reply client/src/App.jsx:12]: Thank you Philip!

**Comment by Whitney-Rene** [comment]
> I realize that all of your components are not taking a prop parameter. One of the key features of React is being able to pass down props and reuse those props in children components. I'm wondering if we can somehow utilize this feature. It could save some redundant API calls to the backend and cleanup some code. Maybe we don't even need to do this? But it's good to think about it.
> 
> Besides this, I think the code looks good overall!

Thank you Philip!  This is good to think about.  Thank you for bringing this to my attention.
I am hopeful that I will have some extra time to go back and begin to think about how to implement this!

**Pull request merged**